sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateHiveTableAsSelectCommand.scala

```diff
@@ -87,8 +101,8 @@ case class CreateHiveTableAsSelectCommand(
       }
     } else {
       try {
-        sparkSession.sessionState.executePlan(InsertIntoTable(
-          metastoreRelation, Map(), query, overwrite = true, ifNotExists = false)).toRdd
+        sparkSession.sessionState.executePlan(InsertIntoTable(metastoreRelation,
```

Comment:
IIUC the partition syntax doesn't contain type, e.g. ```create table t2 using hive partitioned by (c1, c2) as select * from t1```. If one specify `partition by (c1 string, c2 int)` the parser would raise an error, because we have this specified in the parser syntax:

```g4
        createTableHeader ...
        (PARTITIONED BY partitionColumnNames=identifierList)?
        ... #createTable

identifierList
    : '(' identifierSeq ')'
    ;

identifierSeq
    : identifier (',' identifier)*
    ;

```